### PR TITLE
Speed up data in

### PR DESCRIPTION
**Pull request by alan-stokes**
improves on SDP by at least a factor of 2 for Ethernet connected chips and up to 22 times for none ethernet connected chips. 

links to other pull requests that need to be merged in at the same time
- [ ] spynnaker8: https://github.com/SpiNNakerManchester/sPyNNaker8/pull/123
- [ ] spynnaker: https://github.com/SpiNNakerManchester/sPyNNaker/pull/482
- [ ] SPINNMan https://github.com/SpiNNakerManchester/SpiNNMan/pull/113
- [x] SpiNNMachine https://github.com/SpiNNakerManchester/SpiNNMachine/pull/66
- [ ] FrontEndCommon: https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/295
- [ ] GFE: https://github.com/SpiNNakerManchester/SpiNNakerGraphFrontEnd/pull/79

### CODE REVIEW

File: pacman/operations/router_algorithms/basic_dijkstra_routing.py [hunk 1]
```diff
@@ -90,14 +93,18 @@ def __call__(self, placements, machine, machine_graph,
         self._update_all_weights(nodes_info)
 
         # each vertex represents a core in the board
-        progress = ProgressBar(
-            placements.n_placements, "Creating routing entries")
-        for placement in progress.over(placements.placements):
+        progress = None
+        if use_progress_bar:
```
Comment (dkfellows) [inline pacman/operations/router_algorithms/basic_dijkstra_routing.py]: I think I ought to make a dummy progress bar instead which just passes things through.

Comment (alan-stokes) [reply pacman/operations/router_algorithms/basic_dijkstra_routing.py]: possibly, it does open the question about whom should build the progress bar. Given this multi use issue.

Comment (dkfellows) [reply pacman/operations/router_algorithms/basic_dijkstra_routing.py]: SpiNNakerManchester/SpiNNUtils#53 will let this be cleaner

**Comment by dkfellows** [comment]
My comment from earlier has still not been addressed, now that the blocking dependency it involved has been tackled.

**Pull request merged**